downstairs/src/lib.rs

```diff
@@ -599,12 +599,13 @@ where
                     flush_number,
                     gen_number
                 );
+
                 match d
                     .region
                     .region_flush_extent(
                         *extent_id,
-                        *flush_number,
```

Comment:
:scream:

We might want to add a separate type for either gen number of flush number, to prevent this from happening!

Reply:
This was not actually a bug, I was just making function signatures the same
where I did flush/gen/dirty always in that order.  But, yeah, an official type
for flush and/or gen might not be a bad idea to avoid someone flipping them.